src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> l = tail;
+        final Node<T> temp = new Node<>(l, value, null);
+        if (tail == null) {
+            tail = temp;
+            head = temp;
+        } else {
+            tail.next = temp;
+            temp.prev = tail;
+            tail = temp;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        final Node<T> temp = new Node<>(null, value, null);
+        if (index == 0) {
+            temp.next = head;
+            head.prev = temp;
+            head = temp;
+            size++;
+        } else {
+            Node<T> current = getNode(index);
+            temp.prev = current.prev;
+            temp.next = current;
+            current.prev.next = temp;
+            current.prev = temp;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> current = getNode(index);
+        return current.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> current = getNode(index);
+        T temp = current.item;
+        current.item = value;
+        return temp;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = getNode(index);
+        unlink(current);
+        return current.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if ((object == head.item) || (object != null && object.equals(head.item))) {
+            unlink(head);
+            return true;
+        }
+        Node<T> current = head;
+        while (current.next != null
+                && !((current.item == object)
+                || (current.item != null && current.item.equals(object)))) {
+            current = current.next;
+        }
+        if ((current.next == null && object != null)
+                || (object == null && current.equals(tail) && tail.item != null)) {
+            return false;
+        }
+        if (current.equals(tail)) {
+            unlink(tail);
+            return true;
+        }
+        unlink(current);
+        return true;
```

Comment:
```suggestion
        Node<T> current = head;
        while (current.next != null) {
            //check if  current is the right node, if yes -> unlink and return true
            current = current.next;
        }
        return false;
```
I think you should simplify it like this